enrichment-definition/src/main/java/eu/europeana/enrichment/model/impl/TopicEntityImpl.java

```diff
@@ -0,0 +1,235 @@
+package eu.europeana.enrichment.model.impl;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.bson.types.ObjectId;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import dev.morphia.annotations.Entity;
+import dev.morphia.annotations.Id;
+import dev.morphia.annotations.Transient;
+import eu.europeana.enrichment.model.TermEntity;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.model.TopicModel;
+
+
+@Entity(value="TopicEntityImpl")
+public class TopicEntityImpl implements TopicEntity {
+	
+	//id will be used for storing MongoDB _id
+	@Id
+	@JsonIgnore
+	public String _id = new ObjectId().toString();
+	
+	@Transient
+	private TopicModel model;
+	
+	// <{baseurl}/topic/1>
+	private String topicID;
+	
+	// "LDA_EXP1-K15-IT200#TOPIC1"
+	private String identifier;
+	
+	private List<String> labels;
+	// language based description
+	private Map<String,String> descriptions;
+	
+	
+	private List<TermEntity> terms;
+	
+	private List<TermEntity> keywords;
+	
+	private String modelId;
+	
+	private Date created;
+	private Date modified;
+	
+	
+	public TopicEntityImpl()
+	{
+		
+	}
+
+	public TopicEntityImpl(String id, String identifier, List<String> labels, Map<String, String> descriptions,
```

Comment:
This and the constructor below both are probably not needed, so I would remove one of them.

Reply:
the empty construction is needed to avoid compilation errors. The other constructors are needed to create the object from the topic request

Reply:
maybe better remove and use the default constructor and setter methods for building the objects